worlds/yugioh06/logic.py

```diff
@@ -0,0 +1,16 @@
+from typing import Set
+
+from worlds.AutoWorld import LogicMixin
+
+
+class YuGiOh06Logic(LogicMixin):
+    def yugioh06_difficulty(self, player, amount: int):
+        return self.has_group("Core Booster", player, amount)
+
+    def yugioh06_has_individual(self, items: list[str], player: int):
+        amount = 0
+        for item in items:
+            if self.has(item, player):
+                amount += 1
+        return amount
```

Comment:
We added this function in a very recently merged PR! :)

[Core: Add list/item group exclusive methods to CollectionState #3192](https://github.com/ArchipelagoMW/Archipelago/pull/3192)

Understandable if you don't immediately wanna make the jump for current version .apworld compatibility.

If you keep it, though, note my other comment about these functions (LogicMixin is not necessary), and also use `List` from typing instead of `list` for the typing info.